server/src/pages/shot/controller.js

```diff
@@ -88,32 +89,23 @@ exports.launch = function(data) {
   document.dispatchEvent(new CustomEvent("request-addon-present"));
 };
 
-exports.changeShotExpiration = function(shot, expiration) {
-  const wasExpired = model.expireTime !== null && model.expireTime < Date.now();
-  const url = model.backend + "/api/set-expiration";
-  const req = new XMLHttpRequest();
-  req.open("POST", url);
-  req.setRequestHeader("content-type", "application/x-www-form-urlencoded");
-  req.onload = function() {
-    if (req.status >= 300) {
-      const errorMessage = document.getElementById("shotPageAlertErrorUpdatingExpirationTime").textContent;
+exports.changeShotExpiration = async function(shot, expiration) {
+  try {
+    const wasExpired = model.expireTime !== null && model.expireTime < Date.now();
+    const shotUtils = new ShotUtils();
+    await shotUtils.changeShotExpiration(shot, model.backend, model.csrfToken,
+                                         model.defaultExpiration, expiration);
+    model.shot.expireTime = model.expireTime = shot.expireTime;
+    if (wasExpired) {
+      location.reload();
+    }
+    render();
+  } catch (err) {
+    const errorMessage = document.getElementById("shotPageAlertErrorUpdatingExpirationTime").textContent;
+    if (errorMessage) {
```

Comment:
+1, we should check for element instead of message. It's going to error on click of recover as we don't have element there ( that could be a fix in separate issue)